### PR TITLE
Added department and college entities.

### PR DESCRIPTION
**Pull request by Mari-Hayashi**
Created department and college entities.
Modified student and faculty entities so that it stores department as a Department object instead of string.

### CODE REVIEW

File: src/server/README.md [hunk 1]
```diff
@@ -404,13 +410,78 @@ Student API
...
+        -   Creates and saves a department record in the department's table.
+        -   Body:
+            ```
+            {
+                department: {
+                    name: string,
+                    college: {
+                        id: number,
+                        name: string
+					}
```
Comment (Mari-Hayashi) [inline src/server/README.md:457]: This looks unformatted, but this is formatted in my VS.

File: src/server/src/modules/college.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { getRepository } from 'typeorm';
+/**
+ * @description saves a new college in the database
+ * @param name string
+ * @returns Promise
+ */
+export const createcollege = async (
+    name: ICollege['name']
+) => {
+    const college = new College();
```
Comment (johan1505) [inline src/server/src/modules/college.ts]: Is there a reason why here you are creating the college object like this and not how we've been doing it before? Like this:

 const repository = getRepository(College);
 const collegeToInsert = repository.create({
        name,
        departments: [ ],
    });
    return repository.save(collegeToInsert );

File: src/server/src/modules/college.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    college.name = name;
+    college.departments = [];
+    await getRepository(College).save(college);
```
Comment (johan1505) [inline src/server/src/modules/college.ts]: missing return and remove await

File: src/server/src/modules/facultyMember.ts [hunk 4]
```diff
@@ -20,32 +21,41 @@ export const createFacultyMember = (user: IUser) => {
...
         await getRepository(User).update(user.id, {
             biography: user.biography,
             firstName: user.firstName,
             middleName: user.middleName,
             lastName: user.lastName,
         });
-        return await getRepository(FacultyMember).update(id, {
-            departmentId,
+        return await facultyMemberRepository.update(id, {
+            department,
```
Comment (johan1505) [inline src/server/src/modules/facultyMember.ts]: I don't think we need to pass the department here

File: src/server/src/modules/facultyMember.ts [hunk 5]
```diff
@@ -20,32 +21,41 @@ export const createFacultyMember = (user: IUser) => {
...
+
         await getRepository(User).update(user.id, {
             biography: user.biography,
             firstName: user.firstName,
             middleName: user.middleName,
             lastName: user.lastName,
         });
-        return await getRepository(FacultyMember).update(id, {
-            departmentId,
+        return await facultyMemberRepository.update(id, {
```
Comment (johan1505) [inline src/server/src/modules/facultyMember.ts]: remove await

File: src/server/src/modules/department.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+) => {
+    const collegeRepository = getRepository(College);
+    const departmentRepository = getRepository(Department);
+
+    const collegeObject = await collegeRepository.findOne(college.id)
+    if (collegeObject !== undefined) {
+        const department = new Department();
+        department.name = name;
+        department.college = collegeObject;
+        await departmentRepository.save(department);
```
Comment (johan1505) [inline src/server/src/modules/department.ts]: missing return and remove await

**Review by johan1505 — CHANGES_REQUESTED** [review]
Left comments

File: src/server/src/modules/college.ts (hunk 2 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/college.ts]: Done.

File: src/server/src/modules/facultyMember.ts (hunk 4 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/facultyMember.ts]: Done.

File: src/server/src/modules/facultyMember.ts (hunk 5 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/facultyMember.ts]: Done.

File: src/server/src/modules/department.ts (hunk 6 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/department.ts]: Done.

File: src/server/src/modules/college.ts (hunk 3 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/college.ts]: Done.

**Review by johan1505 — APPROVED** [review]

**Pull request merged**